src/common/Cryptography/HMAC_SHA1.h

```diff
@@ -15,23 +15,18 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "SHA1.h"
-#include "BigNumber.h"
-#include "Util.h"
-#include <cstring>
-#include <stdarg.h>
+#ifndef TRINITY_HMAC_SHA1_H
+#define TRINITY_HMAC_SHA1_H
 
-Trinity::Crypto::SHA1::SHA1()
-{
-    SHA1_Init(&_ctx);
-}
+#include "HMAC.h"
+#include <openssl/sha.h>
```

Comment:
```suggestion
#include <openssl/evp.h>
```

Reply:
Actually there is one problem: `SHA_DIGEST_LENGTH` is in `<openssl/sha.h>`
Couple of options:
  * keep `sha.h` (not in favour)
  * define it ourselves
  * use `EVP_MD_size/EVP_MD_CTX_size`

Reply:
The third option isn't an option, these aren't `constexpr`.